pkg/secrets-store/provider_client_test.go

```diff
@@ -200,3 +211,65 @@ func TestMountContentError(t *testing.T) {
 		})
 	}
 }
+
+func TestPluginClientBuilder(t *testing.T) {
+	path, tclean := getTempTestDir(t)
+	defer tclean()
+
+	cb := NewPluginClientBuilder(path)
+	ctx := context.Background()
+
+	for i := 0; i < 10; i++ {
+		server, cleanup := fakeServer(t, path, fmt.Sprintf("server-%d", i))
+		defer cleanup()
+		server.Start()
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		provider := fmt.Sprintf("server-%d", i)
+		go func() {
+			defer wg.Done()
+			if _, err := cb.Get(ctx, provider); err != nil {
+				t.Errorf("Get(%s) = %v, want nil", provider, err)
+			}
```

Comment:
Can we also add a test for concurrent plugin get?

Reply:
Woops added TestPluginClientBuilder_ConcurrentGet